pkg/psmdb/mongos.go

```diff
@@ -0,0 +1,361 @@
+		args = append(args,
+			"--sslMode=preferSSL",
+			"--clusterAuthMode=x509",
+		)
+	}
+
+	if mdSpec.Security != nil && mdSpec.Security.RedactClientLogData {
+		args = append(args, "--redactClientLogData")
+	}
+
+	if msSpec.SetParameter != nil {
+		if msSpec.SetParameter.CursorTimeoutMillis > 0 {
+			args = append(args,
+				"--setParameter",
+				"cursorTimeoutMillis="+strconv.Itoa(msSpec.SetParameter.CursorTimeoutMillis),
+			)
+		}
+	}
+
+	if msSpec.AuditLog != nil && msSpec.AuditLog.Destination == api.AuditLogDestinationFile {
+		if msSpec.AuditLog.Filter == "" {
+			msSpec.AuditLog.Filter = "{}"
+		}
+		args = append(args,
+			"--auditDestination=file",
+			"--auditFilter="+msSpec.AuditLog.Filter,
+			"--auditFormat="+string(msSpec.AuditLog.Format),
+		)
+		switch msSpec.AuditLog.Format {
+		case api.AuditLogFormatBSON:
+			args = append(args, "--auditPath="+MongodContainerDataDir+"/auditLog.bson")
+		default:
+			args = append(args, "--auditPath="+MongodContainerDataDir+"/auditLog.json")
+		}
+	}
+
+	return args
+}
+
+func volumes(cr *api.PerconaServerMongoDB) []corev1.Volume {
+	fvar := false
+	t := true
+	volumes := []corev1.Volume{
+		{
+			Name: InternalKey(cr),
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					DefaultMode: &secretFileMode,
+					SecretName:  InternalKey(cr),
+					Optional:    &fvar,
+				},
+			},
+		},
+		{
+			Name: "ssl",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  cr.Spec.Secrets.SSL,
+					Optional:    &cr.Spec.UnsafeConf,
+					DefaultMode: &secretFileMode,
+				},
+			},
+		},
+		{
+			Name: "ssl-internal",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  cr.Spec.Secrets.SSLInternal,
+					Optional:    &t,
+					DefaultMode: &secretFileMode,
+				},
+			},
+		},
+		{
+			Name: MongodDataVolClaimName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+
+	if *cr.Spec.Mongod.Security.EnableEncryption {
+		volumes = append(volumes,
+			corev1.Volume{
+				Name: cr.Spec.Mongod.Security.EncryptionKeySecret,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						DefaultMode: &secretFileMode,
+						SecretName:  cr.Spec.Mongod.Security.EncryptionKeySecret,
+						Optional:    &fvar,
+					},
+				},
+			},
+		)
+	}
+
+	return volumes
+}
+
+func MongosService(cr *api.PerconaServerMongoDB) corev1.Service {
+	svc := corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-" + "mongos",
+			Namespace: cr.Namespace,
+		},
+	}
+
+	if cr.Spec.Sharding.Mongos != nil {
+		svc.Annotations = cr.Spec.Sharding.Mongos.ServiceAnnotations
+	}
+
+	return svc
+}
+
+func MongosServiceSpec(cr *api.PerconaServerMongoDB) corev1.ServiceSpec {
+	ls := map[string]string{
+		"app.kubernetes.io/component": "mongos",
```

Comment:
please use lables identical to replicaset service
```suggestion
		"app.kubernetes.io/component": "mongos",
		"app.kubernetes.io/name":       "percona-server-mongodb",
		"app.kubernetes.io/instance":   m.Name,
		"app.kubernetes.io/managed-by": "percona-server-mongodb-operator",
		"app.kubernetes.io/part-of":    "percona-server-mongodb",
```